rust/chains/hyperlane-cosmos/src/merkle_tree_hook.rs

```diff
@@ -283,22 +284,12 @@ impl Indexer<MerkleTreeInsertion> for CosmosMerkleTreeHookIndexer {
         &self,
         range: RangeInclusive<u32>,
     ) -> ChainResult<Vec<(Indexed<MerkleTreeInsertion>, LogMeta)>> {
-        let logs_futures: Vec<_> = range
-            .map(|block_number| {
-                let self_clone = self.clone();
-                tokio::spawn(async move {
-                    let logs = self_clone
-                        .indexer
-                        .get_logs_in_block(
-                            block_number,
-                            Self::merkle_tree_insertion_parser,
-                            "MerkleTreeInsertionCursor",
-                        )
-                        .await;
-                    (logs, block_number)
-                })
-            })
-            .collect();
+        let logs_futures = parse_logs_in_range(
```

Comment:
nit: happy either way, but could even take it further and just have all these `fetch_logs_in_range` fns call a single function that also does `execute_and_parse_log_futures` for you too

Reply:
I'll keep it as is for the moment.